{
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2-final"
  },
  "orig_nbformat": 2,
  "kernelspec": {
   "name": "python3",
   "display_name": "Python 3.6.2 64-bit ('python=3.6': conda)",
   "metadata": {
    "interpreter": {
     "hash": "a74011af6f31c2cb45d209a355f58f15c546e74fc60c187994dfbbd79e1a58a7"
    }
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2,
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "from transformers import pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('dimensions-covid19-export-2021-02-22-h18-25-53_publications.csv',\n",
    "                 lineterminator='\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "   Date added  Publication ID                            DOI        PMID  \\\n",
       "0  2021-02-18  pub.1135406536     10.1016/j.jiac.2021.02.012         NaN   \n",
       "1  2021-02-18  pub.1135411657  10.1080/10410236.2021.1886393  33591839.0   \n",
       "2  2021-02-18  pub.1135410507     10.1089/acu.2021.29168.nws         NaN   \n",
       "3  2021-02-18  pub.1135417355            10.1101/mcs.a005967  33593728.0   \n",
       "4  2021-02-18  pub.1135415373     10.21203/rs.3.rs-151871/v1         NaN   \n",
       "\n",
       "        PMCID                                              Title  \\\n",
       "0  PMC7885694  Predictors associated with clinical improvemen...   \n",
       "1         NaN  How Public Health Agencies Break through COVID...   \n",
       "2         NaN                                News from the Field   \n",
       "3         NaN  Blood cancer health disparities in the United ...   \n",
       "4         NaN  The ‘new normal’ includes online prenatal exer...   \n",
       "\n",
       "                                            Abstract  \\\n",
       "0  <sec xmlns=\"https://jats.nlm.nih.gov/ns/archiv...   \n",
       "1  In times of public health emergencies, health ...   \n",
       "2                                                NaN   \n",
       "3  Cancer is a challenging, multifaceted disease ...   \n",
       "4  <p>Background: Prenatal anxiety and depressive...   \n",
       "\n",
       "                            Source title    Source UID Publisher  ...  \\\n",
       "0  Journal of Infection and Chemotherapy  jour.1115587       NaN  ...   \n",
       "1                   Health Communication  jour.1099404       NaN  ...   \n",
       "2                    Medical Acupuncture  jour.1021989       NaN  ...   \n",
       "3                 Molecular Case Studies  jour.1051957       NaN  ...   \n",
       "4                                    NaN           NaN       NaN  ...   \n",
       "\n",
       "            Research Organizations - standardized  \\\n",
       "0             Tokyo Medical and Dental University   \n",
       "1  University of Minnesota; University of Houston   \n",
       "2                                             NaN   \n",
       "3                                             NaN   \n",
       "4                                             NaN   \n",
       "\n",
       "                       GRID IDs  City of Research organization  \\\n",
       "0                 grid.265073.5                          Tokyo   \n",
       "1  grid.17635.36; grid.266436.3           Minneapolis; Houston   \n",
       "2                           NaN                            NaN   \n",
       "3                           NaN                            NaN   \n",
       "4                           NaN                            NaN   \n",
       "\n",
       "  Country of Research organization Funder UIDs of supporting grants  \\\n",
       "0                            Japan    NaN                       NaN   \n",
       "1     United States; United States    NaN                       NaN   \n",
       "2                              NaN    NaN                       NaN   \n",
       "3                              NaN    NaN                       NaN   \n",
       "4                              NaN    NaN                       NaN   \n",
       "\n",
       "  Times cited Altmetric                                     Source Linkout  \\\n",
       "0           0       NaN                                                NaN   \n",
       "1           0       1.0  https://www.tandfonline.com/doi/pdf/10.1080/10...   \n",
       "2           0       NaN                                                NaN   \n",
       "3           0       7.0  http://molecularcasestudies.cshlp.org/content/...   \n",
       "4           0       NaN  https://assets.researchsquare.com/files/rs-151...   \n",
       "\n",
       "                                      Dimensions URL  \n",
       "0  https://app.dimensions.ai/details/publication/...  \n",
       "1  https://app.dimensions.ai/details/publication/...  \n",
       "2  https://app.dimensions.ai/details/publication/...  \n",
       "3  https://app.dimensions.ai/details/publication/...  \n",
       "4  https://app.dimensions.ai/details/publication/...  \n",
       "\n",
       "[5 rows x 31 columns]"
      ],
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Date added</th>\n      <th>Publication ID</th>\n      <th>DOI</th>\n      <th>PMID</th>\n      <th>PMCID</th>\n      <th>Title</th>\n      <th>Abstract</th>\n      <th>Source title</th>\n      <th>Source UID</th>\n      <th>Publisher</th>\n      <th>...</th>\n      <th>Research Organizations - standardized</th>\n      <th>GRID IDs</th>\n      <th>City of Research organization</th>\n      <th>Country of Research organization</th>\n      <th>Funder</th>\n      <th>UIDs of supporting grants</th>\n      <th>Times cited</th>\n      <th>Altmetric</th>\n      <th>Source Linkout</th>\n      <th>Dimensions URL</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>2021-02-18</td>\n      <td>pub.1135406536</td>\n      <td>10.1016/j.jiac.2021.02.012</td>\n      <td>NaN</td>\n      <td>PMC7885694</td>\n      <td>Predictors associated with clinical improvemen...</td>\n      <td>&lt;sec xmlns=\"https://jats.nlm.nih.gov/ns/archiv...</td>\n      <td>Journal of Infection and Chemotherapy</td>\n      <td>jour.1115587</td>\n      <td>NaN</td>\n      <td>...</td>\n      <td>Tokyo Medical and Dental University</td>\n      <td>grid.265073.5</td>\n      <td>Tokyo</td>\n      <td>Japan</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>https://app.dimensions.ai/details/publication/...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>2021-02-18</td>\n      <td>pub.1135411657</td>\n      <td>10.1080/10410236.2021.1886393</td>\n      <td>33591839.0</td>\n      <td>NaN</td>\n      <td>How Public Health Agencies Break through COVID...</td>\n      <td>In times of public health emergencies, health ...</td>\n      <td>Health Communication</td>\n      <td>jour.1099404</td>\n      <td>NaN</td>\n      <td>...</td>\n      <td>University of Minnesota; University of Houston</td>\n      <td>grid.17635.36; grid.266436.3</td>\n      <td>Minneapolis; Houston</td>\n      <td>United States; United States</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>1.0</td>\n      <td>https://www.tandfonline.com/doi/pdf/10.1080/10...</td>\n      <td>https://app.dimensions.ai/details/publication/...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2021-02-18</td>\n      <td>pub.1135410507</td>\n      <td>10.1089/acu.2021.29168.nws</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>News from the Field</td>\n      <td>NaN</td>\n      <td>Medical Acupuncture</td>\n      <td>jour.1021989</td>\n      <td>NaN</td>\n      <td>...</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>https://app.dimensions.ai/details/publication/...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>2021-02-18</td>\n      <td>pub.1135417355</td>\n      <td>10.1101/mcs.a005967</td>\n      <td>33593728.0</td>\n      <td>NaN</td>\n      <td>Blood cancer health disparities in the United ...</td>\n      <td>Cancer is a challenging, multifaceted disease ...</td>\n      <td>Molecular Case Studies</td>\n      <td>jour.1051957</td>\n      <td>NaN</td>\n      <td>...</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>7.0</td>\n      <td>http://molecularcasestudies.cshlp.org/content/...</td>\n      <td>https://app.dimensions.ai/details/publication/...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>2021-02-18</td>\n      <td>pub.1135415373</td>\n      <td>10.21203/rs.3.rs-151871/v1</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>The ‘new normal’ includes online prenatal exer...</td>\n      <td>&lt;p&gt;Background: Prenatal anxiety and depressive...</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>...</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>NaN</td>\n      <td>https://assets.researchsquare.com/files/rs-151...</td>\n      <td>https://app.dimensions.ai/details/publication/...</td>\n    </tr>\n  </tbody>\n</table>\n<p>5 rows × 31 columns</p>\n</div>"
     },
     "metadata": {},
     "execution_count": 3
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "Index(['Date added', 'Publication ID', 'DOI', 'PMID', 'PMCID', 'Title',\n",
       "       'Abstract', 'Source title', 'Source UID', 'Publisher', 'MeSH terms',\n",
       "       'Publication Date', 'PubYear', 'Volume', 'Issue', 'Pagination',\n",
       "       'Open Access', 'Publication Type', 'Authors', 'Corresponding Authors',\n",
       "       'Authors Affiliations', 'Research Organizations - standardized',\n",
       "       'GRID IDs', 'City of Research organization',\n",
       "       'Country of Research organization', 'Funder',\n",
       "       'UIDs of supporting grants', 'Times cited', 'Altmetric',\n",
       "       'Source Linkout', 'Dimensions URL'],\n",
       "      dtype='object')"
      ]
     },
     "metadata": {},
     "execution_count": 53
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "output_type": "error",
     "ename": "KeyError",
     "evalue": "0",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/envs/python=3.6/lib/python3.6/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   2897\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2898\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2899\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.Int64HashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.Int64HashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 0",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-fcc38055093b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'MeSH terms'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'MeSH terms'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misnull\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m==\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/opt/anaconda3/envs/python=3.6/lib/python3.6/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m    880\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    881\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mkey_is_scalar\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 882\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    883\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    884\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mis_hashable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/python=3.6/lib/python3.6/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m_get_value\u001b[0;34m(self, label, takeable)\u001b[0m\n\u001b[1;32m    988\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    989\u001b[0m         \u001b[0;31m# Similar to Index.get_value, but we do not fall back to positional\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 990\u001b[0;31m         \u001b[0mloc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlabel\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    991\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_values_for_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabel\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    992\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/python=3.6/lib/python3.6/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   2898\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2899\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2900\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2901\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2902\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtolerance\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 0"
     ]
    }
   ],
   "source": [
    "df['MeSH terms'][df['MeSH terms'].isnull()==False]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[(df['Abstract'].isnull()==False) & (df['MeSH terms'].isnull()==False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "All model checkpoint layers were used when initializing TFT5Model.\n",
      "\n",
      "All the layers of TFT5Model were initialized from the model checkpoint at t5-small.\n",
      "If your task is similar to the task the model of the checkpoint was trained on, you can already use TFT5Model for predictions without further training.\n",
      "All model checkpoint layers were used when initializing TFT5ForConditionalGeneration.\n",
      "\n",
      "All the layers of TFT5ForConditionalGeneration were initialized from the model checkpoint at t5-small.\n",
      "If your task is similar to the task the model of the checkpoint was trained on, you can already use TFT5ForConditionalGeneration for predictions without further training.\n",
      "[{'summary_text': 'health agencies need to engage and communicate with the public in real-time to share updates and accurate information . this is especially the case for the COVID-19 pandemic where public engagement can potentially save lives and flatten the curve . findings highlight the benefits of strategic social media communication of public health agencies resides in how agencies use social media but also in their formation of network position to amplify their visibility .'}]\n"
     ]
    }
   ],
   "source": [
    "# Initialize the HuggingFace summarization pipeline\n",
    "summarizer = pipeline(\"summarization\")\n",
    "summarized = summarizer(abstract_df[1], min_length=75, max_length=200)\n",
    "\n",
    "# Print summarized text\n",
    "print(summarized)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "\"In times of public health emergencies, health agencies need to engage and communicate with the public in real-time to share updates and accurate information. This is especially the case for the COVID-19 pandemic where public engagement can potentially save lives and flatten the curve. This paper considers how the use of interactive features and strategic network positions of health agencies on social media influenced their public engagement outcomes. Specifically, we analyzed 203 U.S. public health agencies' Twitter activity and the public engagement they received by extracting data from a large-scale Twitter dataset collected from January 21st to May 31st, 2020. Results show that health agencies' network position in addition to their two-way communication strategy greatly influenced the level of public engagement with their COVID-19 related content on Twitter. Findings highlight the benefits of strategic social media communication of public health agencies resides not only in how agencies use social media but also in their formation of network position to amplify their visibility. As official sources of health and risk information, public health agencies should coordinate their social media communication efforts to strategically position themselves in advantageous network positions to augment public engagement outcomes.\""
      ]
     },
     "metadata": {},
     "execution_count": 9
    }
   ],
   "source": [
    "abstract_df[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "zsh: parse error near `()'\n"
     ]
    }
   ],
   "source": [
    "abstract_df.isnull()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "nonnull_abstracts = abstract_df[abstract_df.isnull()==False]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nonnull"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "In times of public health emergencies, health agencies need to engage and communicate with the public in real-time to share updates and accurate information. This is especially the case for the COVID-19 pandemic where public engagement can potentially save lives and flatten the curve. This paper considers how the use of interactive features and strategic network positions of health agencies on social media influenced their public engagement outcomes. Specifically, we analyzed 203 U.S. public health agencies' Twitter activity and the public engagement they received by extracting data from a large-scale Twitter dataset collected from January 21st to May 31st, 2020. Results show that health agencies' network position in addition to their two-way communication strategy greatly influenced the level of public engagement with their COVID-19 related content on Twitter. Findings highlight the benefits of strategic social media communication of public health agencies resides not only in how agencies use social media but also in their formation of network position to amplify their visibility. As official sources of health and risk information, public health agencies should coordinate their social media communication efforts to strategically position themselves in advantageous network positions to augment public engagement outcomes.\n",
      "Cancer is a challenging, multifaceted disease that involves a combination of biological and non-biological factors. Aside from COVID-19, cancer is the second leading cause of death in the United States (U.S.), and the first among Hispanic Americans. The Hispanic population is the largest minority group in the U.S., which is rapidly growing in size. Unfortunately, U.S. Hispanics and other minority groups experience many different health disparities, resulting in poor survival outcomes and a reduced quality of life. Factors such as genomic mutations, lower socioeconomic status, lack of education, reduced access to health care, comorbidities, and environmental factors all contribute to these health care inequalities. In the context of blood cancer health disparities, Hispanic patients are often diagnosed at a younger age and have worse outcomes compared with non-Hispanic individuals. In the present article, we highlight the existing knowledge about cancer health disparities in the Hispanic population, with a focus on chronic and acute leukemia. In our experience at the U.S./Mexico border, analysis of several different blood cancers demonstrated that younger Hispanic patients with acute lymphoid or myeloid leukemia have higher incidence rates and worse prognoses. A combined approach, involving improved health care access and better knowledge of the underlying factors, will allow for more timely diagnoses and the development of intervention strategies aimed at reducing or eliminating the disparities.\n"
     ]
    },
    {
     "output_type": "error",
     "ename": "TypeError",
     "evalue": "cannot concatenate object of type '<class 'list'>'; only Series and DataFrame objs are valid",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-14-6df7dcf207ba>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0msums\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msummarizer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mab\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmin_length\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m75\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmax_length\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m200\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconcat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnonnull_abstracts\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msums\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;31m# Print summarized text\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/python=3.6/lib/python3.6/site-packages/pandas/core/reshape/concat.py\u001b[0m in \u001b[0;36mconcat\u001b[0;34m(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)\u001b[0m\n\u001b[1;32m    282\u001b[0m         \u001b[0mverify_integrity\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mverify_integrity\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    283\u001b[0m         \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 284\u001b[0;31m         \u001b[0msort\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msort\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    285\u001b[0m     )\n\u001b[1;32m    286\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/python=3.6/lib/python3.6/site-packages/pandas/core/reshape/concat.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, objs, axis, join, keys, levels, names, ignore_index, verify_integrity, copy, sort)\u001b[0m\n\u001b[1;32m    357\u001b[0m                     \u001b[0;34m\"only Series and DataFrame objs are valid\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    358\u001b[0m                 )\n\u001b[0;32m--> 359\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    360\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    361\u001b[0m             \u001b[0;31m# consolidate\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: cannot concatenate object of type '<class 'list'>'; only Series and DataFrame objs are valid"
     ]
    }
   ],
   "source": [
    "sums = []\n",
    "\n",
    "for ab in nonnull_abstracts[1:3]:\n",
    "    print(ab)\n",
    "    sums.append(summarizer(ab, min_length=75, max_length=200))\n",
    "\n",
    "\n",
    "# Print summarized text\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "[[{'summary_text': 'health agencies need to engage and communicate with the public in real-time to share updates and accurate information . this is especially the case for the COVID-19 pandemic where public engagement can potentially save lives and flatten the curve . findings highlight the benefits of strategic social media communication of public health agencies resides in how agencies use social media but also in their formation of network position to amplify their visibility .'}], [{'summary_text': 'cancer is the second leading cause of death in the united states . the Hispanic population is the largest minority group in the u.s., which is rapidly growing in size . factors such as genomic mutations, lower socioeconomic status, lack of education, reduced access to health care, comorbidities, and environmental factors all contribute to these health care disparities .'}]]\n"
     ]
    }
   ],
   "source": [
    "print(sums)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "sums=[]\n",
    "doi=[]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_1000 = df.nlargest(1000, 'Times cited')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "________. 110._______\n",
      "________. 120._______\n",
      "________. 130._______\n",
      "________. 140._______\n",
      "________. 150._______\n",
      "________. 160._______\n",
      "________. 170._______\n",
      "________. 180._______\n",
      "________. 190._______\n",
      "________. 200._______\n",
      "________. 210._______\n",
      "________. 220._______\n",
      "________. 230._______\n",
      "________. 240._______\n",
      "________. 250._______\n",
      "________. 260._______\n",
      "________. 270._______\n",
      "________. 280._______\n",
      "________. 290._______\n",
      "________. 300._______\n",
      "________. 310._______\n",
      "________. 320._______\n",
      "________. 330._______\n",
      "________. 340._______\n",
      "________. 350._______\n",
      "________. 360._______\n",
      "________. 370._______\n",
      "________. 380._______\n",
      "________. 390._______\n",
      "________. 400._______\n",
      "________. 410._______\n",
      "________. 420._______\n",
      "________. 430._______\n",
      "________. 440._______\n",
      "________. 450._______\n",
      "________. 460._______\n",
      "________. 470._______\n",
      "________. 480._______\n",
      "________. 490._______\n",
      "________. 500._______\n",
      "________. 510._______\n",
      "________. 520._______\n",
      "________. 530._______\n",
      "________. 540._______\n",
      "________. 550._______\n",
      "________. 560._______\n",
      "________. 570._______\n",
      "________. 580._______\n",
      "________. 590._______\n",
      "________. 600._______\n",
      "________. 610._______\n",
      "________. 620._______\n",
      "________. 630._______\n",
      "________. 640._______\n",
      "________. 650._______\n",
      "________. 660._______\n",
      "________. 670._______\n",
      "________. 680._______\n",
      "________. 690._______\n",
      "________. 700._______\n",
      "________. 710._______\n",
      "________. 720._______\n",
      "________. 730._______\n",
      "________. 740._______\n",
      "________. 750._______\n",
      "________. 760._______\n",
      "________. 770._______\n",
      "________. 780._______\n",
      "________. 790._______\n",
      "________. 800._______\n",
      "________. 810._______\n",
      "________. 820._______\n",
      "________. 830._______\n",
      "________. 840._______\n",
      "________. 850._______\n",
      "________. 860._______\n",
      "________. 870._______\n",
      "________. 880._______\n",
      "________. 890._______\n",
      "________. 900._______\n",
      "________. 910._______\n",
      "________. 920._______\n",
      "________. 930._______\n",
      "________. 940._______\n",
      "________. 950._______\n",
      "________. 960._______\n",
      "________. 970._______\n",
      "________. 980._______\n",
      "________. 990._______\n"
     ]
    }
   ],
   "source": [
    "for i in range(11, 100):\n",
    "    print('________. '+ str(10*i) + '._______' )\n",
    "    sums.append(summarizer(list(df_1000[\"Abstract\"][10*i:10*(i+1)]), min_length=75, max_length=200))\n",
    "    doi.append(df_1000[\"DOI\"][10*i:10*(i+1)])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "sums_flat = [item for sublist in sums for item in sublist]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "sums_list = [item['summary_text'] for item in sums_flat]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "df2 = df_1000.assign(summaryByUs = sums_list)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "Index(['Date added', 'Publication ID', 'DOI', 'PMID', 'PMCID', 'Title',\n",
       "       'Abstract', 'Source title', 'Source UID', 'Publisher', 'MeSH terms',\n",
       "       'Publication Date', 'PubYear', 'Volume', 'Issue', 'Pagination',\n",
       "       'Open Access', 'Publication Type', 'Authors', 'Corresponding Authors',\n",
       "       'Authors Affiliations', 'Research Organizations - standardized',\n",
       "       'GRID IDs', 'City of Research organization',\n",
       "       'Country of Research organization', 'Funder',\n",
       "       'UIDs of supporting grants', 'Times cited', 'Altmetric',\n",
       "       'Source Linkout', 'Dimensions URL', 'summaryByUs'],\n",
       "      dtype='object')"
      ]
     },
     "metadata": {},
     "execution_count": 84
    }
   ],
   "source": [
    "df2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "df3 = df2[['DOI', 'Abstract', 'Title', 'Authors', 'Source title', 'Date added', 'MeSH terms', 'summaryByUs', 'Times cited', 'Source Linkout']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "df4 = df3.rename(columns={\"DOI\": \"doi\", \"Abstract\": \"abstract\", 'Title':'title', 'Authors':'authors', 'Source title':'journal', 'Date added':'date', 'MeSH terms':'concepts', 'Times cited':'citations', 'Source Linkout':'link'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "output_type": "error",
     "ename": "TypeError",
     "evalue": "Column 'concepts' has dtype object, cannot use method 'nlargest' with this dtype",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-98-28f6be825587>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mtop_concepts\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf4\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlargest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m15\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'concepts'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/opt/anaconda3/envs/python=3.6/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mnlargest\u001b[0;34m(self, n, columns, keep)\u001b[0m\n\u001b[1;32m   5683\u001b[0m         \u001b[0mBrunei\u001b[0m      \u001b[0;36m434000\u001b[0m    \u001b[0;36m12128\u001b[0m      \u001b[0mBN\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5684\u001b[0m         \"\"\"\n\u001b[0;32m-> 5685\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0malgorithms\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSelectNFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkeep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkeep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlargest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5686\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5687\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mnsmallest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkeep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"first\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0;34m\"DataFrame\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/python=3.6/lib/python3.6/site-packages/pandas/core/algorithms.py\u001b[0m in \u001b[0;36mnlargest\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1156\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1157\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mnlargest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1158\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"nlargest\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1159\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1160\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mnsmallest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/envs/python=3.6/lib/python3.6/site-packages/pandas/core/algorithms.py\u001b[0m in \u001b[0;36mcompute\u001b[0;34m(self, method)\u001b[0m\n\u001b[1;32m   1273\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_valid_dtype_n_method\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1274\u001b[0m                 raise TypeError(\n\u001b[0;32m-> 1275\u001b[0;31m                     \u001b[0;34mf\"Column {repr(column)} has dtype {dtype}, \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1276\u001b[0m                     \u001b[0;34mf\"cannot use method {repr(method)} with this dtype\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1277\u001b[0m                 )\n",
      "\u001b[0;31mTypeError\u001b[0m: Column 'concepts' has dtype object, cannot use method 'nlargest' with this dtype"
     ]
    }
   ],
   "source": [
    "top_concepts = df4.nlargest(15, 'concepts')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "'2019-nCoV pneumonia was detected in all patients by real-time RT-PCR . 49 (49%) had a history of exposure to the Huanan seafood market . the average age of the patients was 555 years (SD 131), including 67 men and 32 women . INTERPRETATION: 2018-ncoV infection is more likely to affect older males .'"
      ]
     },
     "metadata": {},
     "execution_count": 95
    }
   ],
   "source": [
    "df4.iloc()[5]['summaryByUs']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "320122    [adult, age distribution, aged, covid-19, chin...\n",
       "318440    [adolescent, adult, aged, betacoronavirus, cov...\n",
       "320121    [adult, betacoronavirus, bronchoalveolar lavag...\n",
       "319669    [adult, aged, aged, 80 and over, betacoronavir...\n",
       "317083    [adolescent, adult, age factors, aged, aged, 8...\n",
       "                                ...                        \n",
       "300880    [adult, aged, covid-19, case-control studies, ...\n",
       "302858    [air pollution, betacoronavirus, covid-19, cli...\n",
       "306972    [aged, betacoronavirus, coronavirus infections...\n",
       "308891    [betacoronavirus, covid-19, clinical trials as...\n",
       "221505    [angiotensin-converting enzyme 2, animals, bet...\n",
       "Name: concepts, Length: 1000, dtype: object"
      ]
     },
     "metadata": {},
     "execution_count": 119
    }
   ],
   "source": [
    "df5 = df4\n",
    "import ast\n",
    "df5['concepts']=df5.concepts.astype(str).str.lower().transform(ast.literal_eval)\n",
    "df5['concepts']\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "df5.to_csv('data_w_sums.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "'a recent cluster of pneumonia cases in Wuhan, china, was caused by a novel betacoronavirus . we prospectively collected and analysed data on patients with suspected 2019-nCoV infection by real-time RT-PCR and next-generation sequencing . most of the infected patients were men (30 [73%] of 41; less than half had underlying diseases (13 [32%]), including diabetes (eight [20%]), hypertension (six [15%] and cardiovascular disease ('"
      ]
     },
     "metadata": {},
     "execution_count": 77
    }
   ],
   "source": [
    "sums_list[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "d = {'doi':doi, 'summaryByUs':sums}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "doi_flat = [item for sublist in doi for item in sublist]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       ".3390/v12020135',\n",
       " '10.1053/j.gastro.2020.03.065',\n",
       " '10.1056/nejmc2009191',\n",
       " '10.1056/nejmoa2021680',\n",
       " '10.1111/jth.14869',\n",
       " '10.2214/ajr.20.22976',\n",
       " '10.1001/jamaophthalmol.2020.1291',\n",
       " '10.1148/radiol.2020201160',\n",
       " '10.1126/science.abc6027',\n",
       " '10.1097/rli.0000000000000672',\n",
       " '10.1002/ppul.24718',\n",
       " '10.1126/science.abc1669',\n",
       " '10.1038/s41591-020-0883-7',\n",
       " '10.1148/radiol.2020200823',\n",
       " '10.15585/mmwr.mm6914e1',\n",
       " '10.1136/bmj.m1211',\n",
       " '10.1073/pnas.2004999117',\n",
       " '10.1136/bmj.m1849',\n",
       " '10.1002/jmv.25783',\n",
       " '10.2214/ajr.20.22954',\n",
       " '10.1111/bjd.19163',\n",
       " '10.1371/journal.pone.0231924',\n",
       " '10.1016/s0140-6736(20)31604-4',\n",
       " '10.1002/jmv.25682',\n",
       " '10.1111/jth.14850',\n",
       " '10.1038/s41467-020-16256-y',\n",
       " '10.1016/s0140-6736(20)31182-x',\n",
       " '10.1515/cclm-2020-0369',\n",
       " '10.1172/jci137647',\n",
       " '10.1073/pnas.1922083117',\n",
       " '10.1056/nejmoa2020283',\n",
       " '10.1016/j.envint.2020.105730',\n",
       " '10.1097/cm9.0000000000000774',\n",
       " '10.1097/cm9.0000000000000775',\n",
       " '10.1016/s0140-6736(20)30411-6',\n",
       " '10.1097/cm9.0000000000000722',\n",
       " '10.1093/cid/ciaa198',\n",
       " '10.1007/s11606-020-05762-w',\n",
       " '10.1016/j.ajog.2020.02.017',\n",
       " '10.1056/nejmsa2011686',\n",
       " '10.1084/jem.20200652',\n",
       " '10.1001/jama.2020.10044',\n",
       " '10.1126/science.abc1932',\n",
       " '10.1016/j.molcel.2020.04.022',\n",
       " '10.1002/ddr.21656',\n",
       " '10.1001/jama.2020.4756',\n",
       " '10.1128/jcm.00512-20',\n",
       " '10.1016/j.bbi.2020.05.026',\n",
       " '10.1016/j.cell.2020.04.011',\n",
       " '10.1016/j.immuni.2020.04.023',\n",
       " '10.1126/science.abb7498',\n",
       " '10.1148/radiol.2020200905',\n",
       " '10.1001/jama.2020.10369',\n",
       " '10.1007/s00330-020-06801-0',\n",
       " '10.1016/j.ijid.2020.02.060',\n",
       " '10.1007/s00259-020-04735-9',\n",
       " '10.1016/j.cmet.2020.04.021',\n",
       " '10.1158/2159-8290.cd-20-0422',\n",
       " '10.1021/acsnano.0c02624',\n",
       " '10.1016/j.tmaid.2020.101663',\n",
       " '10.1038/s41586-020-2349-y',\n",
       " '10.1016/s0140-6736(20)31208-3',\n",
       " '10.1016/s2352-4642(20)30095-x',\n",
       " '10.1016/j.chom.2020.03.002',\n",
       " '10.1016/s2215-0366(20)30203-0',\n",
       " '10.1097/ccm.0000000000004363',\n",
       " '10.3390/v12040372',\n",
       " '10.1128/jcm.00310-20',\n",
       " '10.1126/science.abc7520',\n",
       " '10.18632/aging.103000',\n",
       " '10.1038/s41422-020-0305-x',\n",
       " '10.3201/eid2607.200885',\n",
       " '10.15585/mmwr.mm6915e6',\n",
       " '10.1056/nejmp2005492',\n",
       " '10.1093/cvr/cvaa106',\n",
       " '10.1111/his.14134',\n",
       " '10.3785/j.issn.1008-9292.2020.03.03',\n",
       " '10.1001/jama.2020.4861',\n",
       " '10.1056/nejmp2003762',\n",
       " '10.1093/cid/ciaa199',\n",
       " '10.1038/s41586-020-2312-y',\n",
       " '10.1038/s41587-020-0513-4',\n",
       " '10.1001/jama.2020.5893',\n",
       " '10.1073/pnas.2002589117',\n",
       " '10.1056/nejmoa2015301',\n",
       " '10.1016/s0140-6736(20)30920-x',\n",
       " '10.1016/s1473-3099(20)30129-8',\n",
       " '10.1001/jama.2020.1623',\n",
       " '10.1002/jmv.25770',\n",
       " '10.1126/science.abc6284',\n",
       " '10.1186/s40249-020-00662-x',\n",
       " '10.1056/nejmc2003717',\n",
       " '10.1001/jama.2020.2131',\n",
       " '10.1080/22221751.2020.1747363',\n",
       " '10.1126/science.abc3517',\n",
       " '10.1093/cvr/cvaa078',\n",
       " '10.1016/j.autrev.2020.102568',\n",
       " '10.1093/ajcp/aqaa062',\n",
       " '10.1093/infdis/jiaa150',\n",
       " '10.1097/inf.0000000000002660',\n",
       " '10.1016/j.jinf.2020.03.004',\n",
       " '10.1016/j.jmii.2020.02.012',\n",
       " '10.1074/jbc.ac120.013056',\n",
       " '10.1038/s41586-020-2550-z',\n",
       " '10.1056/nejme2001126',\n",
       " '10.1111/jth.14854',\n",
       " '10.1126/science.abb7314',\n",
       " '10.1093/cid/ciaa449',\n",
       " '10.1177/1357633x20916567',\n",
       " '10.1080/22221751.2020.1746199',\n",
       " '10.1111/jth.14859',\n",
       " '10.1126/science.abc2241',\n",
       " '10.1016/s2468-2667(20)30090-6',\n",
       " '10.3390/v12020194',\n",
       " '10.1016/s1473-3099(20)30434-5',\n",
       " '10.1016/s0140-6736(20)30746-7',\n",
       " '10.1001/jamainternmed.2020.2033',\n",
       " '10.1056/nejmc2009166',\n",
       " '10.1016/j.ajp.2020.102083',\n",
       " '10.1038/s41591-020-0916-2',\n",
       " '10.1016/j.psychres.2020.112921',\n",
       " '10.1172/jci138745',\n",
       " '10.1007/s12630-020-01620-9',\n",
       " '10.1111/anae.15054',\n",
       " '10.1126/science.abb8001',\n",
       " '10.3346/jkms.2020.35.e79',\n",
       " '10.1002/jmv.25722',\n",
       " '10.1097/cm9.0000000000000824',\n",
       " '10.1093/jtm/taaa037',\n",
       " '10.1001/jamapediatrics.2020.1467',\n",
       " '10.1542/hpeds.2020-0123',\n",
       " '10.1016/j.jinf.2020.03.041',\n",
       " '10.1002/alr.22579',\n",
       " '10.1016/j.cmi.2020.03.026',\n",
       " '10.3390/ijerph17093165',\n",
       " '10.1164/rccm.202003-0543oc',\n",
       " '10.1148/radiol.2020200843',\n",
       " '10.1016/s2213-2600(20)30243-5',\n",
       " '10.1186/s12967-020-02344-6',\n",
       " '10.1016/s1473-3099(20)30176-6',\n",
       " '10.1007/s12250-020-00207-4',\n",
       " '10.1016/j.ijantimicag.2020.105948',\n",
       " '10.1038/s41586-020-2381-y',\n",
       " '10.1093/infdis/jiaa077',\n",
       " '10.1126/science.abc4776',\n",
       " '10.1016/j.kint.2020.05.006',\n",
       " '10.1016/j.ajogmf.2020.100118',\n",
       " '10.1038/s41586-020-2488-1',\n",
       " '10.1038/s41591-020-0962-9',\n",
       " '10.15252/embj.20105114',\n",
       " '10.1016/j.antiviral.2020.104786',\n",
       " '10.1002/bjs.11627',\n",
       " '10.1016/j.ijid.2020.02.058',\n",
       " '10.1111/apt.15731',\n",
       " '10.1080/22221751.2020.1745095',\n",
       " '10.1016/j.ijid.2020.03.004',\n",
       " '10.1126/science.abb9983',\n",
       " '10.1038/s41379-020-0536-x',\n",
       " '10.1001/jama.2020.6825',\n",
       " '10.1097/rti.0000000000000524',\n",
       " '10.1016/j.micinf.2020.01.004',\n",
       " '10.1016/j.ejim.2020.04.037',\n",
       " '10.1056/nejmc2001899',\n",
       " '10.1172/jci.insight.138999',\n",
       " '10.1148/radiol.2020201365',\n",
       " '10.1002/cbic.202000047',\n",
       " '10.1002/ejhf.1828',\n",
       " '10.1016/j.clim.2020.108427',\n",
       " '10.1111/liv.14435',\n",
       " '10.1001/jama.2020.12603',\n",
       " '10.1126/science.abc1560',\n",
       " '10.1016/j.bbi.2020.03.028',\n",
       " '10.1016/s0140-6736(20)31304-0',\n",
       " '10.3390/ijerph17062032',\n",
       " '10.1056/nejmc2009020',\n",
       " '10.1016/j.ijid.2020.02.033',\n",
       " '10.1016/s1473-3099(20)30162-6',\n",
       " '10.12659/msm.923549',\n",
       " '10.1016/j.scitotenv.2020.138226',\n",
       " '10.12659/msm.923921',\n",
       " '10.1016/j.bbi.2020.04.049',\n",
       " '10.1016/j.scitotenv.2020.138764',\n",
       " '10.1183/13993003.00688-2020',\n",
       " '10.3201/eid2606.200357',\n",
       " '10.1016/j.cytogfr.2020.05.003',\n",
       " '10.1016/j.chom.2020.03.023',\n",
       " '10.1016/j.antiviral.2020.104762',\n",
       " '10.1164/rccm.202002-0445oc',\n",
       " '10.1056/nejmc2011117',\n",
       " '10.1016/s1474-4422(20)30221-0',\n",
       " '10.1161/circulationaha.120.048360',\n",
       " '10.1126/science.abb4489',\n",
       " '10.1016/j.bbrc.2020.03.044',\n",
       " '10.1016/j.jaut.2020.102434',\n",
       " '10.1002/bjs.11746',\n",
       " '10.4014/jmb.2003.03011',\n",
       " '10.1016/j.ijantimicag.2020.105955',\n",
       " '10.1002/jmv.25786',\n",
       " '10.1001/jama.2020.3633',\n",
       " '10.1093/jtm/taaa008',\n",
       " '10.1016/j.envpol.2020.114465',\n",
       " '10.1016/j.cell.2020.08.017',\n",
       " '10.1136/gutjnl-2020-321013',\n",
       " '10.1016/j.ejrad.2020.108961',\n",
       " '10.1016/j.ajog.2020.03.021',\n",
       " '10.1016/s0140-6736(20)30528-6',\n",
       " '10.1016/j.tmrv.2020.02.003',\n",
       " '10.15585/mmwr.mm6924e2',\n",
       " '10.1073/pnas.2004911117',\n",
       " '10.1097/rli.0000000000000670',\n",
       " '10.1126/science.abc6952',\n",
       " '10.1126/science.abc5902',\n",
       " '10.1001/jamacardio.2020.1624',\n",
       " '10.1016/j.it.2020.03.007',\n",
       " '10.1016/j.bbi.2020.03.007',\n",
       " '10.1056/nejmoa2034577',\n",
       " '10.1016/j.jiph.2020.03.019',\n",
       " '10.1001/jamaoto.2020.0780',\n",
       " '10.1016/j.ijid.2020.03.031',\n",
       " '10.1007/s12519-020-00343-7',\n",
       " '10.1080/22221751.2020.1746200',\n",
       " '10.1001/jama.2020.3072',\n",
       " '10.1016/j.jinf.2020.04.002',\n",
       " '10.1016/j.jcv.2020.104362',\n",
       " '10.1159/000507639',\n",
       " '10.1016/j.lfs.2020.117592',\n",
       " '10.1016/j.meegid.2020.104212',\n",
       " '10.1056/nejmp2005689',\n",
       " '10.1016/j.cell.2020.05.042',\n",
       " '10.1111/ajt.15941',\n",
       " '10.1002/jmv.25688',\n",
       " '10.1001/jamacardio.2020.3557',\n",
       " '10.1212/wnl.0000000000009619',\n",
       " '10.1007/s13246-020-00865-4',\n",
       " '10.1016/j.gie.2020.03.019',\n",
       " '10.1136/bmj.m2094',\n",
       " '10.1056/nejmoa2019014',\n",
       " '10.1016/s2213-8587(20)30152-2',\n",
       " '10.1055/s-0040-1710018',\n",
       " '10.1016/j.jns.2020.116832',\n",
       " '10.1371/journal.pone.0230405',\n",
       " '10.5582/bst.2020.01030',\n",
       " '10.1001/jamainternmed.2020.2020',\n",
       " '10.1056/nejmp2015897',\n",
       " '10.1073/pnas.2004978117',\n",
       " '10.2214/ajr.20.22975',\n",
       " '10.7861/clinmed.2019-coron',\n",
       " '10.1073/pnas.2006874117',\n",
       " '10.1002/jmv.25726',\n",
       " '10.1093/jac/dkaa114',\n",
       " '10.1016/j.psychres.2020.112936',\n",
       " '10.1016/j.cell.2020.03.035',\n",
       " '10.1016/s0140-6736(20)31173-9',\n",
       " '10.1001/jamapediatrics.2020.1948',\n",
       " '10.3201/eid2607.200764',\n",
       " '10.1002/lary.28692',\n",
       " '10.1016/j.jmii.2020.03.005',\n",
       " '10.1016/j.scitotenv.2020.138704',\n",
       " '10.1161/circulationaha.120.047549',\n",
       " '10.1001/jama.2020.5046',\n",
       " '10.3201/eid2605.200198',\n",
       " '10.3390/v12020244',\n",
       " '10.1093/cid/ciaa345',\n",
       " '10.1038/s41586-020-2342-5',\n",
       " '10.1016/j.scitotenv.2020.138605',\n",
       " '10.1016/j.jaut.2020.102452',\n",
       " '10.1136/svn-2020-000431',\n",
       " '10.1126/science.abb4557',\n",
       " '10.2196/18810',\n",
       " '10.1001/jama.2020.5227',\n",
       " '10.1001/jama.2020.17023',\n",
       " '10.1038/s41586-020-2588-y',\n",
       " '10.1002/14651858.cd013574',\n",
       " '10.1093/cid/ciaa530',\n",
       " nan,\n",
       " '10.1002/jmv.25825',\n",
       " '10.1056/nejmc2025179',\n",
       " '10.1016/j.bbi.2020.05.048',\n",
       " '10.1016/j.jinf.2020.02.017',\n",
       " '10.1136/bmj.m998',\n",
       " '10.1007/s12519-020-00345-5',\n",
       " '10.1016/s1473-3099(20)30230-9',\n",
       " '10.7150/ijbs.45120',\n",
       " '10.1097/aln.0000000000003296',\n",
       " '10.1093/cid/ciaa225',\n",
       " '10.1126/science.abc8511',\n",
       " '10.1016/j.meegid.2020.104351',\n",
       " '10.1002/alr.22587',\n",
       " '10.7326/m20-1301',\n",
       " '10.1152/ajpheart.00217.2020',\n",
       " '10.1148/radiol.2020201544',\n",
       " '10.1093/cid/ciaa410',\n",
       " '10.1016/j.ijantimicag.2020.105951',\n",
       " '10.1002/bjs.11646',\n",
       " '10.1016/j.scitotenv.2020.138201',\n",
       " '10.1161/hypertensionaha.120.15082',\n",
       " '10.1016/j.lfs.2020.117477',\n",
       " '10.1016/s2468-2667(20)30164-x',\n",
       " '10.1038/s41591-020-0944-y',\n",
       " '10.1007/s12519-020-00354-4',\n",
       " '10.1038/s41586-020-2598-9',\n",
       " '10.1038/s41586-020-2355-0',\n",
       " '10.1038/s41591-020-0824-5',\n",
       " '10.7150/ijbs.45134',\n",
       " '10.1001/jama.2020.8707',\n",
       " '10.1016/j.bbi.2020.04.077',\n",
       " '10.1016/j.compbiomed.2020.103792',\n",
       " '10.1093/cid/ciaa461',\n",
       " '10.3390/ijerph17072381',\n",
       " '10.1111/jgh.15047',\n",
       " '10.3760/cma.j.cn112148-20200220-00105',\n",
       " '10.1073/pnas.2009637117',\n",
       " '10.1002/jmv.25766',\n",
       " '10.1002/jmv.25709',\n",
       " '10.1515/cclm-2020-0285',\n",
       " '10.1158/2159-8290.cd-20-0516',\n",
       " '10.1056/nejmc2007617',\n",
       " '10.1016/s2215-0366(20)30287-x',\n",
       " '10.1016/j.dsx.2020.04.020',\n",
       " '10.1002/jmv.25855',\n",
       " '10.1111/aogs.13867',\n",
       " '10.1126/scitranslmed.abb5883',\n",
       " '10.1002/14651858.cd013652',\n",
       " '10.1038/s41586-020-2404-8',\n",
       " '10.1016/s2214-109x(20)30229-1',\n",
       " '10.1148/radiol.2020201561',\n",
       " '10.1111/jth.14828',\n",
       " '10.1016/j.ajogmf.2020.100107',\n",
       " '10.1038/s41586-020-2571-7',\n",
       " '10.7326/m20-2566',\n",
       " '10.1002/jmv.25915',\n",
       " '10.1126/sciimmunol.abd2071',\n",
       " '10.1001/jama.2020.6266',\n",
       " '10.1016/j.dsx.2020.04.018',\n",
       " '10.1074/jbc.ra120.013679',\n",
       " '10.1016/j.cgh.2020.04.002',\n",
       " '10.7150/ijbs.45053',\n",
       " '10.1186/s40249-020-00640-3',\n",
       " '10.1016/j.annepidem.2020.05.003',\n",
       " '10.1038/s41586-020-2293-x',\n",
       " '10.1016/j.jhep.2020.04.006',\n",
       " '10.1001/jama.2020.6771',\n",
       " '10.1016/j.jinf.2020.04.005',\n",
       " '10.1161/circresaha.120.317055',\n",
       " '10.1016/j.psychres.2020.112915',\n",
       " '10.1056/nejmoa2021756',\n",
       " '10.1001/jama.2020.7197',\n",
       " '10.3390/ijms21072657',\n",
       " '10.1002/art.41285',\n",
       " '10.1148/radiol.2020201237',\n",
       " '10.1136/annrheumdis-2020-217871',\n",
       " '10.1016/j.scitotenv.2020.138878',\n",
       " '10.1002/cpmc.100',\n",
       " '10.1128/jcm.00941-20',\n",
       " '10.1681/asn.2020030276',\n",
       " '10.1371/journal.pone.0230548',\n",
       " '10.1016/j.meegid.2020.104260',\n",
       " '10.1016/j.jcv.2020.104364',\n",
       " '10.1016/j.jinf.2020.03.002',\n",
       " '10.1038/s41564-020-0770-5',\n",
       " '10.1126/science.abd0831',\n",
       " '10.1056/nejmcp2009249',\n",
       " '10.1016/j.chest.2020.04.010',\n",
       " '10.1111/joim.13091',\n",
       " '10.1016/j.jocn.2020.04.062',\n",
       " '10.2807/1560-7917.es.2020.25.9.2000178',\n",
       " '10.1016/j.cell.2020.05.034',\n",
       " '10.1016/j.antiviral.2020.104791',\n",
       " '10.1001/jama.2020.4812',\n",
       " '10.1634/theoncologist.2020-0213',\n",
       " '10.1016/j.chest.2020.03.039',\n",
       " '10.7150/ijbs.45538',\n",
       " '10.1136/bmj.m1435',\n",
       " '10.1056/nejmp2005835',\n",
       " '10.1001/jama.2020.4742',\n",
       " '10.1021/acsnano.0c02823',\n",
       " '10.1038/s41586-020-2608-y',\n",
       " '10.1016/j.scitotenv.2020.138820',\n",
       " '10.1093/cid/ciaa201',\n",
       " '10.2807/1560-7917.es.2020.25.6.2000094',\n",
       " '10.1016/s0140-6736(20)31605-6',\n",
       " '10.1038/s41586-020-2332-7',\n",
       " '10.1055/s-0040-1710050',\n",
       " '10.1016/j.dsx.2020.03.013',\n",
       " '10.1038/s41598-020-76550-z',\n",
       " '10.1136/bmj.m1548',\n",
       " '10.1007/s15010-020-01424-5',\n",
       " '10.1001/jama.2020.2467',\n",
       " '10.3201/eid2606.200239',\n",
       " '10.2807/1560-7917.es.2020.25.12.2000256',\n",
       " '10.1126/science.abd4585',\n",
       " '10.1038/s41577-020-0343-0',\n",
       " '10.1037/hea0000875',\n",
       " '10.1080/22221751.2020.1739565',\n",
       " '10.1093/jtm/taaa031',\n",
       " '10.1016/j.cell.2020.05.032',\n",
       " '10.1056/nejmcp2009575',\n",
       " '10.1016/j.chom.2020.04.017',\n",
       " '10.1148/radiol.2020201343',\n",
       " '10.1093/cid/ciaa243',\n",
       " '10.3346/jkms.2020.35.e112',\n",
       " '10.1007/s40615-020-00756-0',\n",
       " '10.1038/d41587-020-00003-1',\n",
       " '10.1093/jtm/taaa030',\n",
       " '10.1002/bjs.11670',\n",
       " '10.1080/09273948.2020.1738501',\n",
       " '10.1097/jcma.0000000000000270',\n",
       " '10.1007/s40520-020-01570-8',\n",
       " '10.1016/j.micinf.2020.02.006',\n",
       " '10.1016/s2589-7500(20)30026-1',\n",
       " '10.1126/science.abd3871',\n",
       " '10.1136/bmj.m1844',\n",
       " '10.1016/j.tmaid.2020.101738',\n",
       " '10.1111/joim.13089',\n",
       " '10.1038/s41577-020-0321-6',\n",
       " '10.1016/s0140-6736(20)30744-3',\n",
       " '10.1056/nejmoa2022190',\n",
       " '10.1038/s41467-020-17436-6',\n",
       " '10.1016/j.scitotenv.2020.138436',\n",
       " '10.1152/ajpendo.00124.2020',\n",
       " '10.2807/1560-7917.es.2020.25.3.2000044',\n",
       " '10.1016/j.cmi.2020.04.012',\n",
       " '10.1016/j.bbi.2020.04.027',\n",
       " '10.1080/21645515.2020.1735227',\n",
       " '10.1002/jmv.25735',\n",
       " '10.1093/cid/ciaa203',\n",
       " '10.1093/cid/ciaa247',\n",
       " '10.1136/bmj.m1996',\n",
       " '10.1016/j.metabol.2020.154262',\n",
       " '10.1126/sciimmunol.abc3582',\n",
       " '10.1056/nejmc2009567',\n",
       " '10.1016/j.ajem.2020.04.048',\n",
       " '10.3760/cma.j.issn.0254-6450.2020.02.002',\n",
       " '10.1126/science.abd0827',\n",
       " '10.1016/j.ajog.2020.04.014',\n",
       " '10.1016/j.scitotenv.2020.138813',\n",
       " '10.1128/jcm.00461-20',\n",
       " '10.1001/jama.2020.4169',\n",
       " '10.1016/s0140-6736(20)30607-3',\n",
       " '10.3201/eid2606.200301',\n",
       " '10.1002/jmv.25700',\n",
       " '10.1093/cid/ciaa414',\n",
       " '10.1016/j.cca.2020.03.009',\n",
       " '10.1111/jth.14844',\n",
       " '10.1016/j.mayocp.2020.03.024',\n",
       " '10.1111/anae.15071',\n",
       " '10.1080/22221751.2020.1725399',\n",
       " '10.1016/j.diabres.2020.108142',\n",
       " '10.1016/j.cell.2020.07.012',\n",
       " '10.1038/s41586-020-2324-7',\n",
       " '10.1001/jamanetworkopen.2020.8292',\n",
       " '10.1056/nejmc2010418',\n",
       " '10.1016/j.scitotenv.2020.139086',\n",
       " '10.1016/j.immuni.2020.04.003',\n",
       " '10.1002/jmv.25731',\n",
       " '10.1182/blood.2020006520',\n",
       " '10.1093/jamia/ocaa067',\n",
       " '10.2214/ajr.20.23072',\n",
       " '10.1111/all.14364',\n",
       " '10.1056/nejmc2014816',\n",
       " '10.1038/s41586-020-2313-x',\n",
       " '10.1016/j.jcv.2020.104371',\n",
       " '10.1183/13993003.00607-2020',\n",
       " '10.1080/22221751.2020.1732837',\n",
       " '10.1136/bmj.m2107',\n",
       " '10.1016/j.watres.2020.115942',\n",
       " '10.1186/s12967-020-02339-3',\n",
       " '10.1016/j.chest.2020.04.003',\n",
       " '10.1007/s15010-020-01401-y',\n",
       " '10.1038/s41586-020-2548-6',\n",
       " '10.1038/s41586-020-2423-5',\n",
       " '10.1016/s2468-1253(20)30126-6',\n",
       " '10.1016/j.autrev.2020.102523',\n",
       " '10.1002/jmv.25729',\n",
       " '10.15585/mmwr.mm6905e1',\n",
       " '10.3346/jkms.2020.35.e61',\n",
       " '10.1136/bmj.m2516',\n",
       " '10.1007/s10096-020-03913-9',\n",
       " '10.1093/cid/ciaa351',\n",
       " '10.1080/22221751.2020.1743767',\n",
       " '10.1126/science.abc0035',\n",
       " '10.15585/mmwr.mm6919e6',\n",
       " '10.1016/j.autrev.2020.102567',\n",
       " '10.1016/j.cpcardiol.2020.100618',\n",
       " '10.1056/nejmms2009984',\n",
       " '10.1111/ajt.15929',\n",
       " '10.1128/jcm.00557-20',\n",
       " '10.1093/cid/ciaa287',\n",
       " '10.1161/strokeaha.120.030335',\n",
       " '10.7150/ijbs.45472',\n",
       " '10.3855/jidc.12600',\n",
       " '10.12932/ap-200220-0773',\n",
       " '10.1016/j.ijantimicag.2020.105960',\n",
       " '10.1002/jmv.25762',\n",
       " '10.15585/mmwr.mm6909e1',\n",
       " '10.1002/jmv.25742',\n",
       " '10.1001/jama.2020.8279',\n",
       " '10.1093/jamia/ocaa072',\n",
       " '10.1111/bjh.16727',\n",
       " '10.1002/jmv.25889',\n",
       " '10.1016/j.joen.2020.03.008',\n",
       " '10.1073/pnas.2004064117',\n",
       " '10.1007/s00330-020-06817-6',\n",
       " '10.1136/jclinpath-2020-206522',\n",
       " '10.1128/aac.00399-20',\n",
       " '10.1093/eurheartj/ehaa409',\n",
       " '10.1111/all.14309',\n",
       " '10.1126/science.abd4570',\n",
       " '10.1056/nejmoa2026116',\n",
       " '10.1001/jama.2020.8907',\n",
       " '10.1016/s2352-4642(20)30177-2',\n",
       " '10.1007/s00125-020-05180-x',\n",
       " '10.1016/j.ajp.2020.102092',\n",
       " '10.3201/eid2606.200412',\n",
       " '10.1093/cid/ciaa200',\n",
       " '10.1001/jama.2020.2342',\n",
       " '10.1001/jamaneurol.2020.2065',\n",
       " '10.1136/bmj.m1642',\n",
       " '10.3201/eid2607.201595',\n",
       " '10.1056/nejmc2009226',\n",
       " '10.1016/j.jcv.2020.104370',\n",
       " '10.14309/ajg.0000000000000664',\n",
       " '10.1111/ajt.15869',\n",
       " '10.1016/j.dsx.2020.03.011',\n",
       " '10.1017/ice.2020.58',\n",
       " '10.1016/s1473-3099(20)30484-9',\n",
       " '10.1001/jamanetworkopen.2020.11834',\n",
       " '10.1016/j.clim.2020.108448',\n",
       " '10.1002/jmv.25785',\n",
       " '10.1016/j.phrs.2020.104761',\n",
       " '10.3201/eid2606.200516',\n",
       " '10.1016/j.ijid.2020.06.099',\n",
       " '10.1016/j.jaci.2020.05.008',\n",
       " '10.2807/1560-7917.es.2020.25.17.2000257',\n",
       " '10.1016/j.scitotenv.2020.138882',\n",
       " '10.1136/bmj.m1432',\n",
       " '10.1093/qjmed/hcaa089',\n",
       " '10.1042/cs20200163',\n",
       " '10.2214/ajr.20.22969',\n",
       " '10.1016/s2468-2667(20)30133-x',\n",
       " '10.1056/nejmc2013656',\n",
       " '10.1093/cid/ciaa443',\n",
       " nan,\n",
       " '10.1002/jmv.25781',\n",
       " '10.1002/jmv.25767',\n",
       " '10.1056/nejmc2019373',\n",
       " '10.2337/dc20-0576',\n",
       " '10.1016/j.bbi.2020.04.069',\n",
       " '10.1016/j.jinf.2020.04.024',\n",
       " '10.1038/s41581-020-0284-7',\n",
       " '10.1002/jmv.25750',\n",
       " '10.1007/s11655-020-3192-6',\n",
       " '10.1038/s41586-020-2798-3',\n",
       " '10.15585/mmwr.mm6915e5',\n",
       " '10.1111/jce.14479',\n",
       " '10.1016/j.psychres.2020.112958',\n",
       " '10.1016/s2213-2600(20)30121-1',\n",
       " '10.1002/uog.22014',\n",
       " '10.1016/j.jacr.2020.02.008',\n",
       " '10.1016/s2215-0366(20)30308-4',\n",
       " '10.1021/acsnano.0c03252',\n",
       " '10.1097/sla.0000000000003925',\n",
       " '10.1001/jamacardio.2020.1834',\n",
       " '10.1093/brain/awaa240',\n",
       " '10.1016/j.ijid.2020.03.003',\n",
       " '10.3201/eid2605.200146',\n",
       " '10.1038/s41586-020-2607-z',\n",
       " '10.1126/science.abb9789',\n",
       " '10.1371/journal.pone.0233147',\n",
       " '10.1038/s41577-020-0320-7',\n",
       " '10.1007/s11357-020-00186-0',\n",
       " '10.1002/jum.15285',\n",
       " '10.1016/j.lfs.2020.117583',\n",
       " '10.1016/j.molmed.2020.02.008',\n",
       " '10.1186/s12942-020-00202-8',\n",
       " '10.1016/j.intimp.2020.106504',\n",
       " '10.1016/j.scitotenv.2020.138540',\n",
       " '10.1159/000507471',\n",
       " '10.1016/j.ijid.2020.03.013',\n",
       " '10.1002/jmv.25740',\n",
       " '10.1136/bmj.m810',\n",
       " '10.1016/j.jad.2020.08.001',\n",
       " '10.1016/s2214-109x(20)30204-7',\n",
       " '10.1016/j.annonc.2020.04.479',\n",
       " '10.1111/jgs.16472',\n",
       " '10.1056/nejmp2012910',\n",
       " '10.1080/07481187.2020.1748481',\n",
       " '10.4414/smw.2020.20225',\n",
       " '10.1016/j.envres.2020.109819',\n",
       " '10.1186/s13293-020-00304-9',\n",
       " '10.1016/s0140-6736(20)31030-8',\n",
       " '10.1084/jem.20200678',\n",
       " '10.1016/s2214-109x(20)30264-3',\n",
       " '10.1002/jmv.25884',\n",
       " '10.3348/kjr.2020.0146',\n",
       " '10.3201/eid2606.200287',\n",
       " '10.1186/s12916-020-01597-8',\n",
       " '10.1186/s13017-020-00307-2',\n",
       " '10.1002/alr.22577',\n",
       " '10.1038/s41571-020-0362-6',\n",
       " '10.1093/jtm/taaa011',\n",
       " '10.1001/jama.2020.16349',\n",
       " '10.4193/rhin20.114',\n",
       " '10.1007/s11596-020-2172-6',\n",
       " '10.1038/s41591-020-1051-9',\n",
       " '10.1073/pnas.2009799117',\n",
       " '10.1128/aac.00819-20',\n",
       " '10.1038/s41574-020-0364-6',\n",
       " '10.1097/rli.0000000000000674',\n",
       " '10.3760/cma.j.cn121094-20200219-00063',\n",
       " '10.1016/j.ijid.2020.02.050',\n",
       " '10.1038/s41564-020-0771-4',\n",
       " '10.1056/nejmc2015630',\n",
       " '10.1016/j.jhin.2020.04.012',\n",
       " '10.15585/mmwr.mm6912e1',\n",
       " '10.3390/cells9040920',\n",
       " '10.1016/j.jsurg.2020.03.018',\n",
       " '10.7554/elife.57309',\n",
       " '10.1007/s00417-020-04641-8',\n",
       " '10.1038/s41586-020-2639-4',\n",
       " '10.1172/jci140200',\n",
       " '10.1126/sciimmunol.abc8413',\n",
       " '10.1001/jama.2020.10374',\n",
       " '10.1038/s41467-020-16505-0',\n",
       " '10.15585/mmwr.mm6915e4',\n",
       " '10.1111/jsr.13052',\n",
       " '10.7326/m20-1212',\n",
       " '10.2214/ajr.20.22961',\n",
       " '10.1080/01652176.2020.1727993',\n",
       " '10.1016/j.dsx.2020.05.035',\n",
       " '10.3390/nu12041181',\n",
       " '10.1093/eurheartj/ehaa373',\n",
       " '10.1109/rbme.2020.2987975',\n",
       " '10.1016/j.kint.2020.04.002',\n",
       " '10.1002/lary.28672',\n",
       " '10.1111/ajt.15874',\n",
       " '10.1016/j.jinf.2020.03.001',\n",
       " '10.1093/infdis/jiaa119',\n",
       " '10.1056/nejmoa2024671',\n",
       " '10.1128/cmr.00028-20',\n",
       " '10.1016/j.immuni.2020.04.012',\n",
       " '10.1016/s2213-2600(20)30225-3',\n",
       " '10.1001/jama.2020.7233',\n",
       " '10.1055/a-1155-6229',\n",
       " '10.1161/circulationaha.120.047349',\n",
       " '10.1016/j.cgh.2020.04.001',\n",
       " '10.1001/jamaoto.2020.0832',\n",
       " '10.1016/j.medmal.2020.03.007',\n",
       " '10.1001/jamainternmed.2020.3539',\n",
       " '10.1038/s41467-020-16670-2',\n",
       " '10.1016/j.bbi.2020.05.040',\n",
       " '10.3201/eid2608.201274',\n",
       " '10.1021/acsnano.0c02439',\n",
       " '10.1136/gutjnl-2020-321185',\n",
       " '10.1016/j.ijid.2020.03.053',\n",
       " '10.1016/s0140-6736(20)30553-5',\n",
       " '10.1111/jebm.12376',\n",
       " '10.1053/j.gastro.2020.05.032',\n",
       " nan,\n",
       " '10.1080/22221751.2020.1762515',\n",
       " '10.1148/radiol.2020201473',\n",
       " '10.1016/j.jaci.2020.04.027',\n",
       " '10.1183/16000617.0068-2020',\n",
       " '10.1016/j.clnu.2020.03.022',\n",
       " '10.1186/s12992-020-00589-w',\n",
       " '10.15585/mmwr.mm6925a1',\n",
       " '10.1016/s0140-6736(20)30854-0',\n",
       " '10.15585/mmwr.mm6918e1',\n",
       " '10.1093/infdis/jiaa228',\n",
       " '10.1016/j.scitotenv.2020.138277',\n",
       " '10.26355/eurrev_202002_20379',\n",
       " '10.1016/j.cell.2020.06.034',\n",
       " '10.7554/elife.58728',\n",
       " '10.15585/mmwr.mm6932a1',\n",
       " '10.1038/s41587-020-0602-4',\n",
       " '10.5694/mja2.50598',\n",
       " '10.1055/s-0040-1710019',\n",
       " '10.1016/j.jinf.2020.03.060',\n",
       " '10.1002/jmv.25699',\n",
       " '10.1016/j.cell.2020.06.025',\n",
       " '10.1002/path.5471',\n",
       " '10.1111/bjh.16749',\n",
       " '10.1016/j.jmii.2020.03.034',\n",
       " '10.1093/jtm/taaa041',\n",
       " '10.3201/eid2606.200320',\n",
       " '10.12688/f1000research.22457.1',\n",
       " '10.1126/science.abc7424',\n",
       " '10.1093/cid/ciaa539',\n",
       " '10.1136/bmjdrc-2020-001343',\n",
       " '10.1111/ene.14273',\n",
       " '10.1016/j.scitotenv.2020.138835',\n",
       " '10.3346/jkms.2020.35.e149',\n",
       " '10.1016/j.tips.2020.03.006',\n",
       " '10.1126/sciadv.abc5801']"
      ]
     },
     "metadata": {},
     "execution_count": 80
    }
   ],
   "source": [
    "doi_flat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "nonnull_abstracts = nonnull_abstracts.astype('str')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "'Peter Taylor was Editor-in-chief of Stochastic Models for 16\\u2009years; he followed Marcel Neuts, the founding editor, who had been Editor-in-chief from 1985 to 2002. A workshop was held in June 2019, ...'"
      ]
     },
     "metadata": {},
     "execution_count": 52
    }
   ],
   "source": [
    "nonnull_abstracts[14]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "u = df4\n",
    "u['concepts'] = [l.split('; ') for l in u['concepts']]\n",
    "u = u['concepts'].apply(pd.Series).stack().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "                                           keywords  occurences \n",
       "40                                       RNA, Viral           42\n",
       "41                                    Genome, Viral           42\n",
       "42                              Disease Progression           42\n",
       "43                                COVID-19 Vaccines           41\n",
       "44                                    United States           41\n",
       "45                                Treatment Outcome           41\n",
       "46                                            Fever           40\n",
       "47                         Antibodies, Neutralizing           39\n",
       "48                                Infection Control           39\n",
       "49                                        Cytokines           39\n",
       "50                                        Prognosis           38\n",
       "51                                        Incidence           38\n",
       "52                                 Receptors, Virus           36\n",
       "53                                       Prevalence           36\n",
       "54                                            Cough           35\n",
       "55                                        Phylogeny           34\n",
       "56                                     Time Factors           34\n",
       "57                                   Viral Vaccines           34\n",
       "58                                Virus Replication           34\n",
       "59                                            Italy           33\n",
       "60                                       Biomarkers           33\n",
       "61                                  Risk Assessment           32\n",
       "62                          Respiration, Artificial           32\n",
       "63                                          Anxiety           31\n",
       "64              Real-Time Polymerase Chain Reaction           30\n",
       "65                                       Quarantine           30\n",
       "66                             Intensive Care Units           30\n",
       "67                                 Critical Illness           30\n",
       "68                                    Global Health           29\n",
       "69                                  Infant, Newborn           29\n",
       "70                                           Travel           29\n",
       "71  Reverse Transcriptase Polymerase Chain Reaction           29\n",
       "72                                             Mice           27\n",
       "73                                        Pregnancy           27\n",
       "74                                 Health Personnel           27\n",
       "75                            Immunization, Passive           27\n",
       "76                                 Immunoglobulin G           27\n",
       "77                                       Chiroptera           27\n",
       "78                          Cardiovascular Diseases           27\n",
       "79     Middle East Respiratory Syndrome Coronavirus           27"
      ],
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>keywords</th>\n      <th>occurences</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>40</th>\n      <td>RNA, Viral</td>\n      <td>42</td>\n    </tr>\n    <tr>\n      <th>41</th>\n      <td>Genome, Viral</td>\n      <td>42</td>\n    </tr>\n    <tr>\n      <th>42</th>\n      <td>Disease Progression</td>\n      <td>42</td>\n    </tr>\n    <tr>\n      <th>43</th>\n      <td>COVID-19 Vaccines</td>\n      <td>41</td>\n    </tr>\n    <tr>\n      <th>44</th>\n      <td>United States</td>\n      <td>41</td>\n    </tr>\n    <tr>\n      <th>45</th>\n      <td>Treatment Outcome</td>\n      <td>41</td>\n    </tr>\n    <tr>\n      <th>46</th>\n      <td>Fever</td>\n      <td>40</td>\n    </tr>\n    <tr>\n      <th>47</th>\n      <td>Antibodies, Neutralizing</td>\n      <td>39</td>\n    </tr>\n    <tr>\n      <th>48</th>\n      <td>Infection Control</td>\n      <td>39</td>\n    </tr>\n    <tr>\n      <th>49</th>\n      <td>Cytokines</td>\n      <td>39</td>\n    </tr>\n    <tr>\n      <th>50</th>\n      <td>Prognosis</td>\n      <td>38</td>\n    </tr>\n    <tr>\n      <th>51</th>\n      <td>Incidence</td>\n      <td>38</td>\n    </tr>\n    <tr>\n      <th>52</th>\n      <td>Receptors, Virus</td>\n      <td>36</td>\n    </tr>\n    <tr>\n      <th>53</th>\n      <td>Prevalence</td>\n      <td>36</td>\n    </tr>\n    <tr>\n      <th>54</th>\n      <td>Cough</td>\n      <td>35</td>\n    </tr>\n    <tr>\n      <th>55</th>\n      <td>Phylogeny</td>\n      <td>34</td>\n    </tr>\n    <tr>\n      <th>56</th>\n      <td>Time Factors</td>\n      <td>34</td>\n    </tr>\n    <tr>\n      <th>57</th>\n      <td>Viral Vaccines</td>\n      <td>34</td>\n    </tr>\n    <tr>\n      <th>58</th>\n      <td>Virus Replication</td>\n      <td>34</td>\n    </tr>\n    <tr>\n      <th>59</th>\n      <td>Italy</td>\n      <td>33</td>\n    </tr>\n    <tr>\n      <th>60</th>\n      <td>Biomarkers</td>\n      <td>33</td>\n    </tr>\n    <tr>\n      <th>61</th>\n      <td>Risk Assessment</td>\n      <td>32</td>\n    </tr>\n    <tr>\n      <th>62</th>\n      <td>Respiration, Artificial</td>\n      <td>32</td>\n    </tr>\n    <tr>\n      <th>63</th>\n      <td>Anxiety</td>\n      <td>31</td>\n    </tr>\n    <tr>\n      <th>64</th>\n      <td>Real-Time Polymerase Chain Reaction</td>\n      <td>30</td>\n    </tr>\n    <tr>\n      <th>65</th>\n      <td>Quarantine</td>\n      <td>30</td>\n    </tr>\n    <tr>\n      <th>66</th>\n      <td>Intensive Care Units</td>\n      <td>30</td>\n    </tr>\n    <tr>\n      <th>67</th>\n      <td>Critical Illness</td>\n      <td>30</td>\n    </tr>\n    <tr>\n      <th>68</th>\n      <td>Global Health</td>\n      <td>29</td>\n    </tr>\n    <tr>\n      <th>69</th>\n      <td>Infant, Newborn</td>\n      <td>29</td>\n    </tr>\n    <tr>\n      <th>70</th>\n      <td>Travel</td>\n      <td>29</td>\n    </tr>\n    <tr>\n      <th>71</th>\n      <td>Reverse Transcriptase Polymerase Chain Reaction</td>\n      <td>29</td>\n    </tr>\n    <tr>\n      <th>72</th>\n      <td>Mice</td>\n      <td>27</td>\n    </tr>\n    <tr>\n      <th>73</th>\n      <td>Pregnancy</td>\n      <td>27</td>\n    </tr>\n    <tr>\n      <th>74</th>\n      <td>Health Personnel</td>\n      <td>27</td>\n    </tr>\n    <tr>\n      <th>75</th>\n      <td>Immunization, Passive</td>\n      <td>27</td>\n    </tr>\n    <tr>\n      <th>76</th>\n      <td>Immunoglobulin G</td>\n      <td>27</td>\n    </tr>\n    <tr>\n      <th>77</th>\n      <td>Chiroptera</td>\n      <td>27</td>\n    </tr>\n    <tr>\n      <th>78</th>\n      <td>Cardiovascular Diseases</td>\n      <td>27</td>\n    </tr>\n    <tr>\n      <th>79</th>\n      <td>Middle East Respiratory Syndrome Coronavirus</td>\n      <td>27</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "metadata": {},
     "execution_count": 112
    }
   ],
   "source": [
    "u_df = u.value_counts().to_frame(name =\"occurences \")\n",
    "u_df.index.name = 'keywords'\n",
    "u_df.reset_index(inplace=True)\n",
    "u_df[40:80]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ]
}